File: src/leetcode/35. Search Insert Position/search_insert_position35.go
func searchInsert(nums []int, target int) int {
    start := 0
    end := len(nums) - 1

    for start <= end {
        mid := start + ((end - start) / 2)

        if target <= nums[mid] {
            end = mid - 1
        } else {
            start = mid + 1
        }
    }

    return start;
}

File: src/leetcode/35. Search Insert Position/search_insert_position_35.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1

        while start <= end:
            mid = int(start + (end - start))

            if target <= nums[mid]:
                end = mid - 1
            else:
                start = mid + 1

        return start
